### ISSUE TITLE
Modbus protocol

### ISSUE THREAD

**Issue by noyax** [AndyWhittaker/Sofar-HYD6000-ES-ModbusRead#4]
hello, sofarsolar sent me a file with the full modbus protocol used in this inverter. I can send it to you if you want

In french:
bonjour, sofarsolar m'a envoyé un fichier avec le protocole modbus complet utilisé dans cet onduleur. Je peux vous l'envoyer si vous voulez

**Comment by edillmann** [comment]
Bonjour,

Je suis trés interessé par cette documentation, pouvez vous la poster ici dans un pull request ou via un lien externe ?

Merci infiniment

**Comment by paqp** [comment]
@noyax are PV1 and PV2 parameters written there?

**Comment by noyax** [comment]
Yes of course 😉

**Comment by Matthiasfranck** [comment]
Hi @noyax . Would it be possible to receive this file? Would be really helpful to get my connection with sofar inverter.

**Comment by noyax** [comment]
Yes of course
[SofarHYD Modbus protocol.pdf](https://github.com/AndyWhittaker/HYD6000/files/9371555/SofarHYD.Modbus.protocol.pdf)

**Comment by Matthiasfranck** [comment]
@noyax Thnx, this helped me a lot.

BTW, I found something strange in my inverter:

Today load consuming: 8,01kWh
Today generation: 34,47kWh
Today export: 28,72 kWh
Today import: 3,43 kWh

For my understanding:

(generation-export) + import = consumption
9 = 8 ??

I checked the values of import and export and they seem to match exactly with my digital electricity meter. So either the **solar power** is wrong, or the **consumption** is wrong.

My guess is that there is somehow some loss between the total generated power and the actual power that can be used either for export or direct use.

**Comment by noyax** [comment]
Hi, do you have battery? 

in my case the ammeter clamp is not perfectly calibrated and causes a shift in the power imported and exported. (I'm french and use google traduct)

**Comment by Matthiasfranck** [comment]
Hi,

My battery is not yet installed, but it is on its way. With the output of the inverter you will have all battery data needed for HAS.

**Comment by Matthiasfranck** [comment]
No, my battery is not yet installed. So currently a system without battery. As I said, the power imported and exported measured by the inverter is exactly the same as measured by the electricity meter itself.
